drivers/sensor/vcnl4040/Kconfig

```diff
@@ -0,0 +1,168 @@
+# VCNL4040 Proximity and Ambient Light Sensor configuration options
+
+# Copyright (c) 2020 Richard Osterloh
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig VCNL4040
+	bool "VCNL4040 Proximity and Ambient Light Sensor"
+	depends on I2C && HAS_DTS_I2C && HAS_DTS_GPIO
+	help
+	  Enable driver for VCNL4040 sensors.
+
+if VCNL4040
+
+choice
+	prompt "LED current"
+	default VCNL4040_LED_CURRENT_50MA
+
+config VCNL4040_LED_CURRENT_50MA
+	bool "50mA"
+
+config VCNL4040_LED_CURRENT_75MA
+	bool "75mA"
+
+config VCNL4040_LED_CURRENT_100MA
+	bool "100mA"
+
+config VCNL4040_LED_CURRENT_120MA
+	bool "120mA"
+
+config VCNL4040_LED_CURRENT_140MA
+	bool "140mA"
+
+config VCNL4040_LED_CURRENT_160MA
+	bool "160mA"
+
+config VCNL4040_LED_CURRENT_180MA
+	bool "180mA"
+
+config VCNL4040_LED_CURRENT_200MA
+	bool "200mA"
+
+endchoice
+
+choice
+	prompt "LED duty cycle"
+	default VCNL4040_LED_DUTY_CYCLE_40
+
+config VCNL4040_LED_DUTY_CYCLE_40
+	bool "1/40"
+
+config VCNL4040_LED_DUTY_CYCLE_80
+	bool "1/80"
+
+config VCNL4040_LED_DUTY_CYCLE_160
+	bool "1/160"
+
+config VCNL4040_LED_DUTY_CYCLE_320
+	bool "1/320"
+
+endchoice
+
+choice
+	prompt "Proximity integration time"
+	default VCNL4040_PROXIMITY_IT_1T
+
+config VCNL4040_PROXIMITY_IT_1T
+	bool "1T"
+
+config VCNL4040_PROXIMITY_IT_1_5T
+	bool "1.5T"
+
+config VCNL4040_PROXIMITY_IT_2T
+	bool "2T"
+
+config VCNL4040_PROXIMITY_IT_2_5T
+	bool "2.5T"
+
+config VCNL4040_PROXIMITY_IT_3T
+	bool "3T"
+
+config VCNL4040_PROXIMITY_IT_3_5T
+	bool "3.5T"
+
+config VCNL4040_PROXIMITY_IT_4T
+	bool "4T"
+
+config VCNL4040_PROXIMITY_IT_8T
+	bool "8T"
+
+endchoice
+
+choice
+	prompt "Proximity interrupt type"
+	default VCNL4040_PROXIMITY_INT_CLOSE_AWAY
+
+config VCNL4040_PROXIMITY_INT_CLOSE
+	bool "trigger when close"
+
+config VCNL4040_PROXIMITY_INT_AWAY
+	bool "trigger when away"
+
+config VCNL4040_PROXIMITY_INT_CLOSE_AWAY
+	bool "trigger when close or away"
+
+endchoice
+
+config VCNL4040_ENABLE_ALS
+	bool "Enable ambient light sensor"
+	help
+	  Enable Ambient Light Sense (ALS).
+
+choice
+	prompt "ALS integration time"
+	default VCNL4040_ALS_IT_80MS
+
+config VCNL4040_ALS_IT_80MS
+	bool "80ms"
+
+config VCNL4040_ALS_IT_160MS
+	bool "160ms"
+
+config VCNL4040_ALS_IT_320MS
+	bool "320ms"
+
+config VCNL4040_ALS_IT_640MS
+	bool "640ms"
```

Comment:
These should be device tree properties instead of Kconfigs

Reply:
@MaureenHelm thanks for your comments. I have converted the parameters to device tree properties as suggested and added a sample which I have tested on an Adafruit STM32F405 Feather board connected to a VCNL4040 stemmaQT sensor. Hopefully that addresses the buildkite errors